src/canmatrix/canmatrix.py

```diff
@@ -1431,6 +1431,24 @@ def frame_by_id(self, arbitration_id):  # type: (ArbitrationId) -> typing.Union[
                 return test
         return None
 
+    def frame_by_pgn(self, pgn):  # type: (int) -> typing.Union[Frame, None]
+        """Get Frame by pgn (j1939).
+
+        :param int pgn: pgn to search for
+        :rtype: Frame or None
+        """
+
+        if pgn <= 0xEFFF:
```

Comment:
Instead of handle the pgn internals (for me "magic") here, I'd think about to defining some equality check to the `CanId` class itself (to prevent copy paste such `if` in another places).